src/main/java/foxtrot/uniform/charlie/kilo/dao/implementations/UserDaoImpl.java

```diff
@@ -44,7 +44,8 @@ public User add(User user) {
     @Override
     public Optional<User> getByLogin(String email) {
         try (Session session = sessionFactory.openSession()) {
-            User singleResult = session.createQuery("FROM User WHERE email = :email",
+            User singleResult = session.createQuery("FROM User u "
+                            + "LEFT JOIN FETCH u.roles WHERE email = :email",
```

Comment:
```suggestion
                            + "JOIN FETCH u.roles WHERE email = :email",
```
we don't need left join fetch, since user should always have roles